processor/processor.go

```diff
@@ -2160,38 +2173,50 @@ func (proc *HandleT) pipelineWithPause(ctx context.Context, fn func(ctx context.
 	}
 }
 
+//Since mainpipeline func can process jobs in a single batch (when number of jobs are less)
+// and in multiple batches (when number of jobs are more). So, to identify if the received
+//job is a sub-part of a job is a single job. So, `isSplit` variable is needed to be passed
+//accross go routines.
+type subJobT struct {
+	subJobs []*jobsdb.JobT
+	isSplit bool
```

Comment:
In case you follow the configurable `subJobSize` strategy, then you should replace `isSplit` with `subJobCount` here

Reply:
done